stl/src/fiopen.cpp

```diff
@@ -59,12 +63,13 @@ FILE* _Xfiopen(const CharT* filename, ios_base::openmode mode, int prot) {
     mode &= ~(ios_base::ate | ios_base::_Nocreate | ios_base::_Noreplace);
 
     int n = 0;
-    while (valid[n] != 0 && valid[n] != mode) { // look for a valid mode
-        ++n;
-    }
-
-    if (valid[n] == 0) {
-        return nullptr; // no valid mode
+    for (;; ++n) { // look for a valid mode
+        if (valid[n] == mode) {
+            break;
+        }
+        if (valid[n] == 0) {

```

Comment:
Style: We conventionally separate "non-chained" `if` statements with a newline. This makes it easier to see when we have independent tests, versus an `if` chained to `else if`.